diff/walking/differ.go

```diff
@@ -56,6 +57,38 @@ func NewWalkingDiff(store content.Store) diff.Comparer {
 	}
 }
 
+func ReadonlyMounts(mounts []mount.Mount) []mount.Mount {
```

Comment:
The function seems misnamed if this is a NOP for non-overlay mounts

Reply:
Agree. I made changes, now the overlay-specific code is in `readonlyOverlay` and added some more logic to `readonlyMounts` to deal with other mount types.